eventstore/test_views.py

```diff
@@ -387,6 +394,161 @@ def test_postbirth_other_passport_origin(self):
         self.assertEqual(registration.passport_country, "other")
 
 
+class MessagesViewSetTests(APITestCase):
+    url = reverse("messages-list")
+
+    def generate_hmac_signature(self, data, key):
+        data = JSONRenderer().render(data)
+        h = hmac.new(key.encode(), data, sha256)
+        return base64.b64encode(h.digest()).decode()
+
+    def test_successful_messages_request(self):
+        """
+        Should create a new Messages object in the database
+        """
+        user = get_user_model().objects.create_user("test")
+        user.user_permissions.add(Permission.objects.get(codename="add_messages"))
+        self.client.force_authenticate(user)
+        data = {
+            "messages": [
+                {
+                    "id": "9e12d04c-af25-40b6-aa4f-57c72e8e3f91",
+                    "from": "sender-wa-id",
+                    "timestamp": "1518694700",
+                    "type": "image",
+                    "message_direction": "I",
+                    "recipient_type": "type",
+                    "context": {
+                        "from": "sender-wa-id-of-context-message",
+                        "group_id": "group-id-of-context-message",
+                        "id": "message-id-of-context-message",
+                        "mentions": ["wa-id1", "wa-id2"],
+                    },
+                    "image": {
+                        "file": "absolute-filepath-on-coreapp",
+                        "id": "media-id",
+                        "link": "link-to-image-file",
+                        "mime_type": "media-mime-type",
+                        "sha256": "checksum",
+                        "caption": "image-caption",
+                    },
+                    "location": {
+                        "address": "1 Hacker Way, Menlo Park, CA, 94025",
+                        "name": "location-name",
+                    },
+                    "system": {"body": "system-message-content"},
+                    "text": {"body": "text-message-content"},
```

Comment:
The webhook API format changes often, so I think we should harden our API against this.

In this test, we could add some `"random": "data",`, and ensure that it gets placed inside the `data` field on our model, to simulate WhatsApp adding a field to the request that we're not expecting.

Reply:
okay, thanks